### PR TITLE
Switch to ESLint from JSHint+JSCS

### LINKED ISSUE #1533: Replace jshint+jscs with eslint
JSCS is now [unmaintained](http://eslint.org/blog/2016/07/jscs-end-of-life), in order to provide full compatibility with ESLint, ESLint team [identifying](https://github.com/eslint/eslint/issues/5858) most popular JSCS presets and providing for them ESLint replacement.

Grunt config is one of them, sharable config for it is now published - https://github.com/markelog/eslint-config-grunt

Would you like to switch to it?

### PR DESCRIPTION
**Pull request by markelog**
Fixes #1533

### CODE REVIEW

File: lib/grunt/task.js [hunk 1]
```diff
@@ -1,5 +1,11 @@
 'use strict';
 
+// Keep track of the number of log.error() calls.
+var errorcount,
+
+  // The last specified tasks message.
+  lastInfo;
```
Comment (markelog) [inline lib/grunt/task.js]: This is a difference between JSCS and ESLint `one-var`/`disallowMultipleVarDecl` rules, i figure it is easier to change the code then to propose changes in ESLint

File: package.json [hunk 2]
```diff
@@ -56,10 +56,10 @@
   },
   "devDependencies": {
     "difflet": "~0.2.3",
-    "grunt-contrib-jshint": "~1.0.0",
+    "eslint-config-grunt": "^1.0.1",
     "grunt-contrib-nodeunit": "~0.4.1",
     "grunt-contrib-watch": "~1.0.0",
-    "grunt-jscs": "~2.8.0",
+    "grunt-eslint": "^18.0.0",
```
Comment (markelog) [inline package.json]: `19.0.0` is not used since it doesn't supports node < 4, which means you can't use ESLint 3.0, we can apply same approach as in jquery repo though - https://github.com/jquery/jquery/commit/030191ae32dfcb7ecb2efb61d17a4964a3633e44

Comment (markelog) [inline package.json]: But if you do, i would suggest to do this in separate commit

File: lib/grunt/task.js (hunk 1 above)
Comment (shama) [inline lib/grunt/task.js]: Are you saying ESLint doesn't allow multiple var statements?

@cowboy has [some good opinions](http://benalman.com/news/2012/05/multiple-var-statements-javascript/) on that. So Grunt would definitely prefer to use multiple var statements.

Comment (markelog) [inline lib/grunt/task.js]: With 
```js
"one-var": [
"one-var": [
  "error",
  {
    "uninitialized": "always",
    "initialized": "never"
  }
]
```

It doesn't allow initialized values with different `var` keywords. I believe it satisfy the idea in the blog post you mentioned (see "In summary" at the end) - 

> My rule is as-follows: I’ll put multiple comma-separated declarations in a single var statement, but only if they don’t have assignments. 

```js
// SIMPLE AND SEXY

var foo, bar, baz;
var abc = 1;
var def = 2;
```

In any case, if you want me to change this rule or omit it, I can do that too

Comment (shama) [inline lib/grunt/task.js]: Ah cool. Thanks for pointing that out. Could we put the declarations onto one line and just concat the code comments with `and`?

File: package.json (hunk 2 above)
Comment (vladikoff) [inline package.json]: yeah we can do that!

File: lib/grunt/task.js (hunk 1 above)
Comment (markelog) [inline lib/grunt/task.js]: done

**Comment by vladikoff** [comment]
@markelog awesome, thanks so much!

**Comment by markelog** [comment]
@vladikoff wait a bit plz, one more change coming up

**Comment by vladikoff** [comment]
👍

**Comment by markelog** [comment]
Oh, you already merged :)

**Comment by vladikoff** [comment]
@markelog yea you can send another PR, no worriES!

**Comment by markelog** [comment]
Ok, cool

**Comment by markelog** [comment]
Never mind, new changes are blocked by the https://github.com/eslint/eslint/issues/3458
